src/rpc/mining.cpp

```diff
@@ -341,7 +341,11 @@ UniValue getmininginfo(const UniValue& params, bool fHelp)
     obj.pushKV("currentblocksize", (uint64_t)nLastBlockSize);
     obj.pushKV("currentblocktx",   (uint64_t)nLastBlockTx);
     obj.pushKV("difficulty",       (double)GetNetworkDifficulty());
-    obj.pushKV("errors",           GetWarnings("statusbar"));
+    UniValue wobj(UniValue::VOBJ);
+    auto warnings = GetWarnings("statusbar");
+    wobj.pushKV("timestamp", warnings.first);
+    wobj.pushKV("msg", warnings.second);
+    obj.pushKV("errors",           wobj);
```

Comment:
The "errors" key should stay the same for compatibility.

```suggestion
    auto warnings = GetWarnings("statusbar");
    obj.pushKV("errors",           warnings.second);
    obj.pushKV("errorstimestamp",  warnings.first);
```